docs/source/make_a_release.rst

```diff
@@ -179,3 +179,44 @@ Step-by-step
       rm -rf dist/* && \
       rmvirtualenv pynwb-${release}-release && \
       rmvirtualenv pynwb-${release}-install-test
+
+-------------
+Conda Release
```

Comment:
My understanding is that Conda is the package manager, and conda-forge is where the packages are hosted and served from

Reply:
Could you also:
* rename `Step-by-step` -> `PyPI: Step-by-step`
* rename `Conda Release` -> `Conda: Step-by-step`

Also also a dedication section `GitHub: Step-by-step`  (this is the upload as GitHub releases)